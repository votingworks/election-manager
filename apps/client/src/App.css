/* Allow selectors to be used multiple times in this file for grouping of concerns. */
/* stylelint-disable no-duplicate-selectors */

/* Reset */
html {
  box-sizing: border-box;
  line-height: 1;
  user-select: none;
}
*,
*::before,
*::after {
  box-sizing: inherit;
}

body {
  margin: 0;
}

table {
  border-collapse: collapse;
}

fieldset {
  margin: 0;
  border: none;
  padding: 0;
}

legend {
  display: block;
}

img {
  display: block;
}

:link,
:visited {
  color: rgb(0, 0, 238);
}

:focus {
  /* outline: rgb(77, 144, 254) dashed 0.25rem; */
  outline: none;
}

/* Accessibility Utilities */
.screen-reader-only {
  position: absolute !important; /* stylelint-disable-line declaration-no-important */
  width: 1px;
  height: 1px;
  overflow: hidden;
  clip: rect(1px, 1px, 1px, 1px);
  clip-path: polygon(0 0, 0 0, 0 0);
}

/* stylelint-disable-next-line selector-class-pattern */
html body .cvox_indicator_container {
  /* Extra specificity/!important ensure this rule is followed. */
  display: none !important; /* stylelint-disable-line declaration-no-important */
}
