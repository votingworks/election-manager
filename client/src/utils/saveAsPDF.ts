import { Election } from '@votingworks/ballot-encoder'

export default async function saveAsPDF(
  fileNamePrefix: string,
  election: Election,
  precinctName = 'all-precincts'
): Promise<boolean> {
  const data = await window.kiosk!.printToPDF()
  const fileWriter = await window.kiosk!.saveAs({
    defaultPath: `${`${fileNamePrefix}-${election.county.name}-${election.title}-${precinctName}`
      .replace(/[^a-z0-9]+/gi, '-')
      .replace(/(^-|-$)+/g, '')
      .toLocaleLowerCase()}.pdf`,
  })
  if (!fileWriter) {
    return false
  }
  fileWriter.write(data)
  await fileWriter.end()
  return true
}
