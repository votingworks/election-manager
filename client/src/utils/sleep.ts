const sleep = (ms = 1000) => new Promise((resolve) => setTimeout(resolve, ms))

export default sleep
