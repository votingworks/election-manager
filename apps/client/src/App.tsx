import React from 'react'
import { BrowserRouter, Route } from 'react-router-dom'

import './App.css'

import { LocalStorage, KioskStorage } from './utils/Storage'

import AppRoot, { Props as AppRootProps, AppStorage } from './AppRoot'

export interface Props {
  storage?: AppRootProps['storage']
}

const defaultStorage = window.kiosk
  ? new KioskStorage<AppStorage>()
  : new LocalStorage<AppStorage>()

const App = ({ storage = defaultStorage }) => (
  <BrowserRouter>
    <Route
      path="/"
      render={(props) => <AppRoot storage={storage} {...props} />}
    />
  </BrowserRouter>
)

export default App
