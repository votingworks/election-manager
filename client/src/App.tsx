import React from 'react'
import { BrowserRouter, Route } from 'react-router-dom'

import './App.css'

import { LocalStorage } from './utils/Storage'

import AppRoot, { Props as AppRootProps, AppStorage } from './AppRoot'

export interface Props {
  storage?: AppRootProps['storage']
}

const App = ({ storage = new LocalStorage<AppStorage>() }) => (
  <BrowserRouter>
    <Route
      path="/"
      render={(props) => <AppRoot storage={storage} {...props} />}
    />
  </BrowserRouter>
)

export default App
