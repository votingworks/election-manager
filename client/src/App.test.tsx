import React from 'react'

import { fireEvent, render, screen } from '@testing-library/react'
import { Election } from '@votingworks/ballot-encoder'
import { MemoryStorage } from './utils/Storage'
import {
  AppStorage,
  electionStorageKey,
  configuredAtStorageKey,
} from './AppRoot'

import fakeKiosk from '../test/helpers/fakeKiosk'

import App from './App'

import sleep from './utils/sleep'

import eitherNeitherElectionUntyped from '../test/fixtures/eitherneither-election.json'

const eitherNeitherElection = (eitherNeitherElectionUntyped as unknown) as Election

jest.mock('./components/HandMarkedPaperBallot')

beforeEach(() => {
  // we don't care about network errors logged to the console, they're crowding things
  jest.spyOn(console, 'error').mockImplementation(() => {}) // eslint-disable-line @typescript-eslint/no-empty-function

  window.location.href = '/'
  window.kiosk = fakeKiosk()
})

it('create election works', async () => {
  const { getByText, getAllByText } = render(<App />)

  await screen.findByText('Create New Election Definition')
  fireEvent.click(getByText('Create New Election Definition'))
  await screen.findByText('Ballots')

  fireEvent.click(getByText('Ballots'))
  fireEvent.click(getAllByText('View Ballot')[0])
  fireEvent.click(getByText('English/Spanish'))
})

it('basic navigation works', async () => {
  // mock the election we want
  const storage = new MemoryStorage<AppStorage>()
  await storage.set(electionStorageKey, eitherNeitherElection)
  await storage.set(configuredAtStorageKey, new Date().toISOString())
  const { container, getByText, getAllByText, queryAllByText } = render(
    <App storage={storage} />
  )

  await screen.findByText('0 official ballots')

  // go print some ballots
  fireEvent.click(getByText('Export Ballot Package'))
  expect(window.kiosk?.saveAs).toHaveBeenCalledTimes(1)

  // we're not mocking the filestream yet
  await screen.findByText('Download Failed')

  fireEvent.click(getByText('Ballots'))
  fireEvent.click(getAllByText('View Ballot')[0])
  fireEvent.click(getByText('Precinct'))
  fireEvent.click(getByText('Absentee'))
  fireEvent.click(getByText('Test'))
  fireEvent.click(getByText('Official'))
  fireEvent.click(getByText('Print 1 Official', { exact: false }))
  expect(window.kiosk?.print).toHaveBeenCalledTimes(1)

  // this is ugly but necessary for now to wait just a bit for the data to be stored
  await sleep(0)

  fireEvent.click(getByText('Ballots'))
  getByText('1 official ballot', { exact: false })
  fireEvent.click(getByText('Printed Ballots Report'))
  fireEvent.click(queryAllByText('Print Report')[0])
  expect(window.kiosk?.print).toHaveBeenCalledTimes(2)

  fireEvent.click(getByText('Tally'))
  fireEvent.click(getByText('Print Test Decks'))
  getByText('Chester')
  getByText('District 5')
  fireEvent.click(getByText('All Precincts'))
  await screen.findByText('Generating Test Deck...')

  await screen.findByText('Print Test Deck')
  fireEvent.click(getByText('Print Test Deck'))
  expect(container).toMatchSnapshot()
  expect(window.kiosk?.print).toHaveBeenCalledTimes(3)

  fireEvent.click(getByText('Tally'))
  fireEvent.click(getByText('View Test Ballot Deck Tally'))
  fireEvent.click(getByText('All Precincts'))
  await screen.findByText('Print Results Report')
  expect(container).toMatchSnapshot()
  fireEvent.click(getByText('Print Results Report'))
  expect(window.kiosk?.print).toHaveBeenCalledTimes(4)

  fireEvent.click(getByText('Definition'))
  fireEvent.click(getByText('JSON Editor'))

  // remove the election
  fireEvent.click(getByText('Remove'))
  fireEvent.click(getByText('Remove Election Definition'))

  await screen.findByText('Configure Election Manager')
})
