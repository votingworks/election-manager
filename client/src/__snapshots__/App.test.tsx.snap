// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic navigation works 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="sc-AxjAm isOMPC"
  >
    <main
      class="sc-AxirZ dUjkQS"
    >
      <div
        class="sc-AxiKw gTDzBu"
      >
        <div
          class="sc-fzokOt cibPwK"
        >
          <h1>
            Test Deck
          </h1>
          <p>
            <strong>
              Election:
            </strong>
             
            Mock General Election Choctaw 2020
            <br />
            <strong>
              Precinct:
            </strong>
             
            All Precincts
          </p>
          <p>
            <button
              class="sc-fzoLsD dftQGP"
              type="button"
            >
              Print Test Deck
            </button>
          </p>
          <p>
            <button
              class="sc-fzoLsD cIOEoD"
              role="option"
              type="button"
            >
              back to Test Deck list
            </button>
          </p>
        </div>
      </div>
    </main>
    <div
      class="sc-AxhCb lobKPH"
    >
      <div
        class="sc-AxhUy fjWzfB"
      >
        <div
          class="sc-AxgMl bFBkDf"
        >
          VxPaper
        </div>
        <div
          class="sc-AxheI fniENP"
        >
          Election Manager
        </div>
      </div>
      <div
        class="sc-Axmtr jFWbcr"
      >
        <button
          class="sc-fzoLsD ivZRNV"
          role="option"
          type="button"
        >
          Definition
        </button>
        <button
          class="sc-fzoLsD ivZRNV active-section"
          role="option"
          type="button"
        >
          Ballots
        </button>
        <button
          class="sc-fzoLsD cIOEoD"
          role="option"
          type="button"
        >
          Tally
        </button>
      </div>
      <div
        class="sc-AxmLO jrWrYg"
      >
        <p
          class="sc-fznyAO dooajl"
        >
          No USB
        </p>
      </div>
    </div>
  </div>
  <div
    class="print-only"
  >
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      5
      , precinct 
      6522
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      5
      , precinct 
      6522
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      5
      , precinct 
      6522
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6524
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6524
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6524
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6525
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6525
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6525
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6526
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6526
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6526
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6527
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6527
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6527
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6528
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6528
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6528
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6529
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6529
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6529
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6532
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6532
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6532
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6534
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6534
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6534
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6536
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6536
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      3
      , precinct 
      6536
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6537
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6537
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      1
      , precinct 
      6537
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6538
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6538
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      4
      , precinct 
      6538
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6539
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6539
      .
    </div>
    <div>
      <h1>
        Mocked HMPB
      </h1>
      Election: 
      Mock General Election Choctaw 2020
      <br />
      Ballot Style 
      2
      , precinct 
      6539
      .
    </div>
  </div>
  <div />
</div>
`;

exports[`basic navigation works 2`] = `
<div>
  <div
    class="sc-AxjAm isOMPC"
  >
    <main
      class="sc-AxirZ dUjkQS"
    >
      <div
        class="sc-AxiKw gTDzBu"
      >
        <div
          class="sc-fzokOt cibPwK"
        >
          <h1>
            Test Ballot Deck Tally
          </h1>
          <p>
            <strong>
              Election:
            </strong>
             
            Mock General Election Choctaw 2020
            <br />
            <strong>
              Precinct:
            </strong>
             
            All Precincts
          </p>
          <p>
            <button
              class="sc-fzoLsD dftQGP"
              type="button"
            >
              Print Results Report
            </button>
          </p>
          <p>
            <button
              class="sc-fzoLsD cIOEoD"
              role="option"
              type="button"
            >
              back to Test Deck list
            </button>
          </p>
        </div>
      </div>
    </main>
    <div
      class="sc-AxhCb lobKPH"
    >
      <div
        class="sc-AxhUy fjWzfB"
      >
        <div
          class="sc-AxgMl bFBkDf"
        >
          VxPaper
        </div>
        <div
          class="sc-AxheI fniENP"
        >
          Election Manager
        </div>
      </div>
      <div
        class="sc-Axmtr jFWbcr"
      >
        <button
          class="sc-fzoLsD ivZRNV"
          role="option"
          type="button"
        >
          Definition
        </button>
        <button
          class="sc-fzoLsD ivZRNV"
          role="option"
          type="button"
        >
          Ballots
        </button>
        <button
          class="sc-fzoLsD cIOEoD active-section"
          role="option"
          type="button"
        >
          Tally
        </button>
      </div>
      <div
        class="sc-AxmLO jrWrYg"
      >
        <p
          class="sc-fznyAO dooajl"
        >
          No USB
        </p>
      </div>
    </div>
  </div>
  <div
    class="print-only"
  >
    <div
      class="sc-fznWOq dLNTMU"
    >
      <h1>
        Test Ballot Deck Tally
      </h1>
      <p>
        <strong>
          Election:
        </strong>
         
         Mock General Election Choctaw 2020
        <br />
        <strong>
          Precinct:
        </strong>
         
        All Precincts
      </p>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              39 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            United States
            , 
            President
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Presidential Electors for Joe Biden for President and Kamala Harris for Vice President
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Presidential Electors for Donald J. Trump for President and Michael R. Pence for Vice President
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Presidential Electors for Phil Collins for President and Bill Parker for Vice President
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              39 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            United States
            , 
            Senate 
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Mike Espy
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Cindy Hyde-Smith
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Jimmy Edwards
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              39 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            US House Of Representatives
            , 
            1st Congressional District
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Antonia Eliason
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  26
                </td>
              </tr>
              <tr>
                <td>
                  Trent Kelly
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              39 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            Northern District
            , 
            Supreme Court District 3(Northern) Position 3
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Josiah Dennis Coleman
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  26
                </td>
              </tr>
              <tr>
                <td>
                  Percy L. Lynchard
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              9 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            Election Commissioner 01
            , 
            Election Commissioner  01
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Glynda Chaney Fulce
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  9
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              9 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            Election Commissioner 02
            , 
            Election Commissioner 02
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Charles Beck
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  6
                </td>
              </tr>
              <tr>
                <td>
                  Sharon Brooks
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  3
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              9 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            Election Commissioner 03
            , 
            Election Commissioner 03
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Dorothy Anderson
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  9
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              9 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            Election Commissioner 04
            , 
            Election Commissioner 04
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Willie Mae Guillory
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  6
                </td>
              </tr>
              <tr>
                <td>
                  Lewis Wright
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  3
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              3 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            Election Commissioner 05
            , 
            Election Commissioner 05
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Ouida A Loper
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  2
                </td>
              </tr>
              <tr>
                <td>
                  Wayne McLeod
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  1
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              3 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            School Board District 5
            , 
            School Board 05
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  Michael D Thomas
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  3
                </td>
              </tr>
              <tr>
                <td>
                  Write-In
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              0 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            State Of Mississippi
            , 
            Ballot Measure 1 – Either/Neither
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  yes
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  26
                </td>
              </tr>
              <tr>
                <td>
                  no
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              0 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            State Of Mississippi
            , 
            Ballot Measure 1 – Pick One
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  yes
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  26
                </td>
              </tr>
              <tr>
                <td>
                  no
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              39 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            State Of Mississippi
            , 
            Ballot Measure 2: House Concurrent Resolution No. 47
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  yes
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  26
                </td>
              </tr>
              <tr>
                <td>
                  no
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="sc-fznMAR xKWvD"
      >
        <div
          class="sc-fzokOt bcrqaN"
        >
          <div
            class="sc-fznxKY bOVfyb ignore-prose"
          >
            <span
              class="sc-fznyAO drvcLn"
            >
              39 ballots
               cast /
               
              0 overvotes
               /
               
              0 undervotes
            </span>
          </div>
          <h3>
            State Of Mississippi
            , 
            Ballot Measure 3: House Bill 1796 - Flag Referendum
          </h3>
          <table
            class="sc-fzqNJr gbfqYQ"
          >
            <tbody>
              <tr>
                <td>
                  yes
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  26
                </td>
              </tr>
              <tr>
                <td>
                  no
                </td>
                <td
                  class="sc-fzoyAV hqQnKY"
                >
                  13
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
